android/src/main/kotlin/io/customer/customer_io/CustomerIoPlugin.kt

```diff
@@ -142,9 +154,28 @@ class CustomerIoPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
         }
     }
 
+    private fun registerDeviceToken(params: Map<String, Any>) {
+        val token = params.getString(Keys.Tracking.TOKEN)
+        CustomerIO.instance().registerDeviceToken(token)
+    }
+
+    private fun trackMetric(params: Map<String, Any>) {
+        val deliveryId = params.getString(Keys.Tracking.DELIVERY_ID)
+        val deliveryToken = params.getString(Keys.Tracking.DELIVERY_TOKEN)
+        val event = MetricEvent.getEvent(params.getProperty<String>(Keys.Tracking.METRIC_EVENT))
+
+        if (event == null) {
+            logger.info("metric event type isn't correct")
+            return
+        }
+
+        CustomerIO.instance().trackMetric(
+            deliveryID = deliveryId, deviceToken = deliveryToken, event = event
+        )
+    }
+
     private fun setDeviceAttributes(params: Map<String, Any>) {
-        val attributes =
-            params.getProperty<Map<String, Any>>(Keys.Tracking.ATTRIBUTES) ?: emptyMap()
+        val attributes = params.getProperty<Map<String, Any>>(Keys.Tracking.ATTRIBUTES) ?: return
```

Comment:
Since it's relevant to this PR, I wanted to put a comment here. 

This is an example of using the guard design pattern in Kotlin 😄